blockchain/v0/reactor.go

```diff
@@ -366,6 +366,18 @@ FOR_LOOP:
 			err := state.Validators.VerifyCommitLight(
 				chainID, firstID, first.Height, second.LastCommit)
 
+			var stateMachineValid bool
+			if err == nil {
```

Comment:
[nit] since `stateMachineValid` isn't used outside the scope of the following block, it can be moved inside the block. This change reduces the scope of `stateMachineValid`

```suggestion
			if err == nil {
				var stateMachineValid bool
```

Reply:
Yes good point